### PR TITLE
Add support for DPAD Center key of Android TV remote controller

### LINKED ISSUE #77077: DPAD Center key of Android TV remote controller is detected as "Uknown" key, not a specific key.
### Godot version

4.0.2

### System information

Android TV (1080p) API 29(Android 10), Google Chromecast(Android 12)

### Issue description

Pressing or releasing a DPAD Center key of Android TV remote controller is detected as
an InputEventKey of "Uknown" key, not a specific key.
The cause is that there is no key mapping of Android dpad center key.

I tried writing a patch adding key mapping for Android DPAD center.
https://github.com/hrk4649/godot_fork/tree/support_android_dpad_center
I added some lines after referencing how to deal with "JIS_KANA" in the godot code,
as a similar key to Android DPAD center.
This patch looks working but I am not sure if it is correct solution. 

### Steps to reproduce

Using the reproduction project zip file below, 

1 setup a project to export Android.
    export_presets.cfg is like following.
    Setting release keystore is required.

```
gradle_build/use_gradle_build=true
architectures/armeabi-v7a=true
architectures/arm64-v8a=true
architectures/x86=true
architectures/x86_64=true
keystore/release="XXX"
keystore/release_user="XXX"
keystore/release_password="XXX"
package/unique_name="xxx.InputEventSample"
```

2 prepare a device which is a real device or an Android TV emulator.
3 run the project on the device.
4 press DPAD center key of a remote controller. Input events will be displayed on the screen or Logcat.
Following output is expected.

```
InputEventKey: keycode=8388607 (Unknown), mods=none, physical=false, pressed=true, echo=false
InputEventKey: keycode=8388607 (Unknown), mods=none, physical=false, pressed=false, echo=false
```

### Minimal reproduction project

[input_event_sample.zip](https://github.com/godotengine/godot/files/11473067/input_event_sample.zip)

### PR DESCRIPTION
**Pull request by hrk4649**
This PR fixes #77077

### CODE REVIEW

File: platform/android/android_keys_utils.h [hunk 1]
```diff
@@ -60,6 +60,7 @@ static AndroidGodotCodePair android_godot_code_pairs[] = {
 	{ AKEYCODE_DPAD_DOWN, Key::DOWN }, // (20) Directional Pad Down key.
 	{ AKEYCODE_DPAD_LEFT, Key::LEFT }, // (21) Directional Pad Left key.
 	{ AKEYCODE_DPAD_RIGHT, Key::RIGHT }, // (22) Directional Pad Right key.
+	{ AKEYCODE_DPAD_CENTER, Key::DPAD_CENTER }, // (23) Directional Pad Center key.
```
Comment (m4gr3d) [inline platform/android/android_keys_utils.h:63]: Seems like we should be able to reuse `Key::ENTER` for `AKEYCODE_DPAD_CENTER` instead?

That would mirror what we are doing for the other dpad keycodes.

**Review by m4gr3d — CHANGES_REQUESTED** [review]

File: platform/android/android_keys_utils.h (hunk 1 above)
Comment (AThousandShips) [reply platform/android/android_keys_utils.h:63]: Especially unless we add it to standard actions, I'd expect the controller to be able to trigger `ui_accept` and such

File: core/os/keyboard.h [hunk 2]
```diff
@@ -167,7 +167,6 @@ enum class Key {
 	KEYBOARD = SPECIAL | 0x71,
 	JIS_EISU = SPECIAL | 0x72,
 	JIS_KANA = SPECIAL | 0x73,
-
```
Comment (AThousandShips) [inline core/os/keyboard.h:170]: This space should be left if it was there originally I'd say

Comment (hrk4649) [reply core/os/keyboard.h]: @AThousandShips 
Thank you for your advice. I fixed it.

**Review by AThousandShips — APPROVED** [review]
LGTM, simple and straightforward

**Review by m4gr3d — APPROVED** [review]
Looks good!
Can you squash the commits.

**Comment by akien-mga** [comment]
I squashed the commits myself, should be good to merge.

**Comment by akien-mga** [comment]
Thanks! And congrats for your first merged Godot contribution :tada:

**Comment by YuriSizov** [comment]
Cherry-picked for 4.0.4.

**Comment by akien-mga** [comment]
Cherry-picked for 3.6.

**Comment by akien-mga** [comment]
Doesn't cherry-pick cleanly for 3.5.x, there have been too many Android input changes which seem risky to backport to a stable branch, so I'll leave this out.

I note that DPAD Center has a mapping in 3.5 already, but with another key:
```
        { KEY_PERIODCENTERED, AKEYCODE_DPAD_CENTER },
```

**Pull request merged**